assembl/views/api2/votes.py

```diff
@@ -196,3 +200,56 @@ def vote_results_csv(request):
     ctx._instance.csv_results(output, histogram)
     output.seek(0)
     return Response(body_file=output, content_type='text/csv')
+
+
+@view_config(context=InstanceContext, request_method='GET',
+             ctx_instance_class=VotingWidget,
+             name="vote_results_csv", permission=P_DISC_STATS)
+def global_vote_results_csv(request):
+    ctx = request.context
+    user_id = request.authenticated_userid
+    if not user_id:
+        raise HTTPUnauthorized
+    widget = ctx._instance
+    if widget.activity_state != "ended":
+        permissions = get_permissions(user_id, ctx.get_discussion_id())
+        if P_ADMIN_DISC not in permissions:
+            raise HTTPUnauthorized()
+    user_prefs = LanguagePreferenceCollection.getCurrent()
+    ideas = widget.db.query(Idea
+        ).join(AbstractIdeaVote, AbstractIdeaVote.idea_id==Idea.id
+        ).join(AbstractVoteSpecification
+        ).filter(AbstractVoteSpecification.widget_id==widget.id
+        ).distinct().all()
+    idea_ids = [i.id for i in ideas]
+    titles = [(idea.title.best_lang(user_prefs).value, idea.id) for idea in ideas]
```

Comment:
Not defensive enough. `title` can be `None`.

Reply:
You got to be kidding me... That should almost be made non-null in the DB.

Reply:
But I did do a defensive version.